src/pybind/mgr/dashboard/services/ceph_service.py

```diff
@@ -167,20 +167,36 @@ def send_command(cls, srv_type, prefix, srv_spec='', **kwargs):
     @staticmethod
     def _get_smart_data_by_device(device):
         if 'daemons' in device and device['daemons']:
-            daemons = [daemon for daemon in device['daemons'] if daemon.startswith('osd')]
+            daemons = [
+                daemon for daemon in device['daemons']
+                if daemon.startswith('osd')
+            ]
             if daemons:
-                svc_type, svc_id = daemons[0].split('.')
-                dev_smart_data = CephService.send_command(
-                    svc_type, 'smart', svc_id, devid=device['devid'])
-                for dev_id, dev_data in dev_smart_data.items():
-                    if 'error' in dev_data:
-                        logger.warning(
-                            '[SMART] error retrieving smartctl data for device ID "%s": %s', dev_id,
-                            dev_data)
+                dev_smart_data = None
+                for daemon in daemons:
```

Comment:
I'm not too fond of iterating all OSDs blindly here until we hit one accidentally that is capable of returning the requested data, because it isn't down. I'd prefer to find out which OSDs are in a state that would allow them to return the requested data and use one of them (or iterate them all).